src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -459,8 +459,15 @@ def _update_learning_rates(
             if config.interval == interval and current_idx % config.frequency == 0:
                 monitor_val = None
                 if config.reduce_on_plateau:
-                    # If instance of ReduceLROnPlateau, we need a monitor
                     monitor_key = config.monitor
+                    if monitor_key is None:
+                        raise RuntimeError(
```

Comment:
I think this is not necessary. We can assume it is not None in case of ReduceOnPlateau. It is getting checked [here](https://github.com/Lightning-AI/lightning/blob/61c28cb428a13c2aea6d7f3f55e0f00431a4ea4e/src/pytorch_lightning/core/optimizer.py#L289-L294) prior to training. 

If we still think it should raise here, let's make the message the same. Btw, we can't add doc links to error messages reliably, because they are not permanent.

Reply:
You're right. This is only for letting mypy know `monitor_key` is not `None`.
If it's okay to add `assert monitor_key is not None`, I'll revise the PR to do so. 👍

Reply:
[fd44cfc](https://github.com/Lightning-AI/lightning/pull/13555/commits/fd44cfc03d61b126664c645888b645663bccb3fd)

Reply:
Revised in fd44cfc. :pray: